### ISSUE TITLE
VB: IDE crashes when typing code to initialise array of tuple with delegate member.

### ISSUE THREAD

**Issue by ericmutta** [dotnet/roslyn#21785]
**Version Used**: 

Visual Studio 2017 Community 15.3.2

**Steps to Reproduce**:

Add a VB console project. The objective is to type the following code:

```vb

Module Module1
  Private mTupleArray As (X As Integer, P As Func(Of Byte(), Integer))() = {
             (X:=0, P:=Nothing), (X:=0, P:=AddressOf MyFunction)
          }

  Sub Main()
  End Sub

  Public Function MyFunction(ArgBytes As Byte()) As Integer
  End Function
End Module
```
However you can never type and finish the array initialiser portion here:

```vb
  Private mTupleArray As (X As Integer, P As Func(Of Byte(), Integer))() = {
             (X:=0, P:=Nothing), (X:=0, P:=AddressOf MyFunction)
          }
``` 
Everything goes OK up until this point where you have typed `AddressOf` but haven't reached to `MyFunction`:
![vs_tuples_crash1](https://user-images.githubusercontent.com/20465797/29813378-8accdf32-8cb2-11e7-8fcb-37175286c079.png)

Things are still OK when you start typing `MyFunction` and Intellisense kicks in to complete it for you:
![vs_tuples_crash2](https://user-images.githubusercontent.com/20465797/29813481-e2d92f00-8cb2-11e7-9778-a775bab08024.png)

Once you hit tab to have Intellisense complete `MyFunction`, a brief moment later the IDE crashes:
![vs_tuples_crash3](https://user-images.githubusercontent.com/20465797/29813558-16b87cf4-8cb3-11e7-8a3f-ca94cd54c9e8.png)

I tried initialising without using the tuple member names, and it still crashes:
![vs_tuples_crash4](https://user-images.githubusercontent.com/20465797/29813701-8bbae618-8cb3-11e7-976c-29fee50b2322.png)

I also tried using the initialiser on a single line (this is the real code I was writing when I discovered the crash):
![vs_tuples_crash5](https://user-images.githubusercontent.com/20465797/29814598-949966d0-8cb6-11e7-81e7-41dcba457d59.png)

I think that `AddressOf MyFunction` expression is the culprit when used in an array initialiser.

**Expected Behavior**:

It should not crash (the code is semantically and syntactically correct)...in fact this works just fine:

```vb
  Sub Main()
    mTupleArray(1) = (X:=0, P:=AddressOf MyFunction) '<-------- no crash here.
  End Sub
```

**Actual Behavior**:

As demonstrated above: IDE crashes.

**Comment by jcouv** [comment]
Thanks for the detailed bug report. The issue is confirmed as late as version 15.4 preview 1.
I have a crash dump file and I'm investigating it.

**Comment by ericmutta** [comment]
> @jcouv I have a crash dump file and I'm investigating it.

Much appreciated Julien!  Tuples are new so there are likely to be a few rough edges (see also issue [#21812](https://github.com/dotnet/roslyn/issues/21812) which has two other crash scenarios involving tuples)...hopefully many will be resolved by 15.4!
